### PR TITLE
UGENE-5371. Problem with opening assembly from folder with non-ASCII symbols in path

### PR DESCRIPTION
**Pull request by yalgaer**
The solution is to always pass a file handle (int) instead of file name (const char*) to samtools while working with BAM/SAM files

### CODE REVIEW

File: src/corelibs/U2Formats/src/BAMUtils.cpp [hunk 1]
```diff
@@ -463,28 +522,54 @@ bool BAMUtils::hasValidFastaIndex(const GUrl& fastaUrl) {
...
+ * Builds and saves index for BAM file. Returns 0 if the index was created correctly.
+ * Exact copy of 'bam_index_build2' with a correct unicode file names support.
+ */
+static int bam_index_build_unicode(const QString& bamFileName) {
+    FILE* bFile = BAMUtils::openFile(bamFileName, "rb");
+    CHECK(bFile != nullptr, -1);
+    bamFile fp = bam_dopen(fileno(bFile), "rb");
+    if (fp == nullptr) {
+        closeFileIfOpen(bFile);
+        fprintf(stderr, "[bam_index_build2] fail to open the BAM file.\n");
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Formats/src/BAMUtils.cpp:535]: Why don't you use the UGENE standard logging?

File: src/corelibs/U2Formats/src/BAMUtils.cpp [hunk 2]
```diff
@@ -463,28 +522,54 @@ bool BAMUtils::hasValidFastaIndex(const GUrl& fastaUrl) {
...
+        return -1;
+    }
+    FILE* fpidx = BAMUtils::openFile(bamFileName + ".bai", "wb");
+    if (fpidx == nullptr) {
+        fprintf(stderr, "[bam_index_build2] fail to create the index file.\n");
+        return -1;
+    }
+    bam_index_save(idx, fpidx);
+    bam_index_destroy(idx);
+    fclose(fpidx);
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Formats/src/BAMUtils.cpp:552]: Has a return value, which could be an error, need to check it

**Review by DmitriiSukhomlinov — CHANGES_REQUESTED** [review]
Minor comments

File: src/corelibs/U2Formats/src/BAMUtils.cpp (hunk 1 above)
Comment (yalgaer) [reply src/corelibs/U2Formats/src/BAMUtils.cpp:535]: This method is a copy-paste of the original samtools method, so I preserved as many original lines as possible (our samtools logs to stderr)

File: src/corelibs/U2Formats/src/BAMUtils.cpp (hunk 2 above)
Comment (yalgaer) [reply src/corelibs/U2Formats/src/BAMUtils.cpp:552]: Same here: this is the original behavior of samtools.

**Pull request merged**